llvm/lib/SYCLLowerIR/LowerWGLocalMemory.cpp

```diff
@@ -118,7 +145,7 @@ static bool allocaWGLocalMemory(Module &M) {
 
 PreservedAnalyses SYCLLowerWGLocalMemoryPass::run(Module &M,
                                                   ModuleAnalysisManager &) {
-  if (allocaWGLocalMemory(M))
-    return PreservedAnalyses::none();
-  return PreservedAnalyses::all();
+  bool Changed = inlineAllocateLocalMemoryFunc(M);
+  Changed |= allocaWGLocalMemory(M);
+  return Changed ? PreservedAnalyses::none() : PreservedAnalyses::all();
```

Comment:
Is it too conservative to use `::none()` here?

Reply:
I'm not sure which analysis we can preserve here